### ISSUE TITLE
Auto-correction of Style/SymbolProc introduces ArgumentError: no receiver given

### ISSUE THREAD

**Issue by tstatter** [rubocop/rubocop#12962]
#### Summary:
When RuboCop applies the auto-correction for the `Style/SymbolProc` cop to a specific use case, it results in a `ArgumentError: no receiver given`. This happens when applying this cop to a `before_destroy` active record callback.

#### Steps to reproduce:
Consider this piece of code:

```ruby
before_destroy :ensure_deletion_request_present, unless: proc { |record| record.skip_deletion_request_check }
```

When RuboCop applies auto-correction for the `Style/SymbolProc` cop, it changes the above to:

```ruby
before_destroy :ensure_deletion_request_present, unless: proc(&:skip_deletion_request_check)
```

Now running the `destroy` method throws the `ArgumentError: no receiver given`.

#### Expected behavior:
The `Style/SymbolProc` autocorrection should not introduce this error. This specific case might require some consideration where a proc requires the receiver. 

#### Actual behavior:
The auto-correction changes the code in a way that results in a `ArgumentError: no receiver given` during runtime.

#### RuboCop version:
rubocop-1.64.0

1.64.0 (using Parser 3.3.1.0, rubocop-ast 1.31.3, running on ruby 3.2.0) [x86_64-linux]

**Comment by koic** [comment]
RuboCop is a static analysis tool for the Ruby language. The issue is caused by behavior that is extended by Active Support. You can switch to behavior that corresponds to the extensions provided by Active Support by using rubocop-rails or setting `AllCops:ActiveSupportExtensionsEnabled: true`:

```yaml
# .rubocop.yml
AllCops:
  ActiveSupportExtensionsEnabled: true
```

https://docs.rubocop.org/rubocop/1.64/cops_style.html#allcopsactivesupportextensionsenabled-false-default-3

**Comment by tstatter** [comment]
Active Support is written in the Ruby language, so having auto correct that breaks Ruby code doesn't smell great but Ok thanks.

**Comment by bbatsov** [comment]
> RuboCop is a static analysis tool for the Ruby Programming Language.

What Koic said is that we can't really know with static analysis what libraries people are using exactly just by looking at the code - therefore the configuration flag and the separate `rubocop-rails` extensions which assumes people are using Rails.

**Comment by tstatter** [comment]
I appreciate it's on us to use the right extensions,  on the flips side,  for inexperienced or naive users this autocorrect can create very hard to find bugs, so was just letting you know.
